--- security/selinux/selinuxfs.c.orig	2026-05-27 16:11:48.237605510 +0800
+++ security/selinux/selinuxfs.c	2026-05-27 17:38:25.085598420 +0800
@@ -238,13 +238,32 @@ static int sel_open_handle_status(struct
 static ssize_t sel_read_handle_status(struct file *filp, char __user *buf,
 				      size_t count, loff_t *ppos)
 {
-	struct page    *status = filp->private_data;
+	struct page *status = filp->private_data;
+	char *tmpbuf;
+	ssize_t ret;
+	bool is_app_zygote = false;
 
 	BUG_ON(!status);
 
-	return simple_read_from_buffer(buf, count, ppos,
-				       page_address(status),
-				       sizeof(struct selinux_kernel_status));
+	if (strstr(current->comm, "app_zygote") != NULL)
+		is_app_zygote = true;
+
+	if (is_app_zygote) {
+		tmpbuf = kmalloc(PAGE_SIZE, GFP_KERNEL);
+		if (!tmpbuf)
+			return -ENOMEM;
+		memcpy(tmpbuf, page_address(status), PAGE_SIZE);
+		((struct selinux_kernel_status *)tmpbuf)->sequence = 1;
+		((struct selinux_kernel_status *)tmpbuf)->policyload = 0;
+		ret = simple_read_from_buffer(buf, count, ppos, tmpbuf,
+					      sizeof(struct selinux_kernel_status));
+		kfree(tmpbuf);
+		return ret;
+	} else {
+		return simple_read_from_buffer(buf, count, ppos,
+					       page_address(status),
+					       sizeof(struct selinux_kernel_status));
+	}
 }
 
 static int sel_mmap_handle_status(struct file *filp,
